### ISSUE TITLE
Rails/PluralizationGrammar false positive for -1 (or is at least just inconsistent)

### ISSUE THREAD

**Issue by DMA57361** [bbatsov/rubocop#2412]
The new `Rails/PluralizationGrammar` in 0.35.0 is being a little inconsistent over whether -1 day/week/etc should be plural or singular. Personally, I prefer the singular over the plural, and this is what the cop allows _most_ of the time.

For example, given the following file:
```
Timecop.travel(-1.week)
Timecop.travel(- 1.week)
Timecop.travel(Time.now.utc-1.week)
Timecop.travel(Time.now.utc - 1.week)
```
We get the following result:
```
$ rubocop example.rb --only Rails/PluralizationGrammar
Inspecting 1 file
C

Offenses:

example.rb:1:16: C: Rails/PluralizationGrammar: Prefer -1.weeks.
Timecop.travel(-1.week)
               ^^^^^^^

1 file inspected, 1 offense detected
```

**Comment by rrosenblum** [comment]
Great question. Whether `-1` should be plural or singular came up in conversation when `Rails/PluralizationGrammer` was being built. The comment is one an outdated diff in #2316. There seems to be a fair bit of debate as to whether `-1` should be singular or plural (http://english.stackexchange.com/questions/9735/is-1-singular-or-plural). It seems to boil down to what is the connotation. `(-1.days)` is plural because it is a group. `- (1.day)` is singular because you are subtracting a single day. 

I am not expert on linguistics, but based on the example given, I would say the cop is functioning properly. 

If enough people feel strongly about `-1` being singular instead of plural, we can consider adding a configuration to force it to be one or the other.

**Comment by DMA57361** [comment]
That the cop didn't consider the first two the same (`-1.week` vs `- 1.week`) is what made me consider this possibly a bug. The suppose the latter is arguably a bit off since it has a useless space in it, but does seem odd that the cop parses the two differently.

**Comment by rrosenblum** [comment]
> does seem odd that the cop parses the two differently.

At first, I was going to say the same thing, but they are different.

`Timecop.travel(-1.week)` and `Timecop.travel(- 1.week)` seem to be relying on a little ruby magic to work. `-` and `+` are not defined on Kernel, but must have a global implementation to imply `0 -` and `0 +`. Another way of thinking about it would be `-1.week` is `-1.send(:week)` and `- 1.week` is `send(:-, 1.send(:week))`. 

`Timecop.travel(Time.now.utc-1.week)` and `Timecop.travel(Time.now.utc - 1.week)` are the same thing. Think of it as `Time.now.utc.send(:-, 1.week)`. Technically, `1.week` is positive. We are subtracting a whole unit from `now`.

I hope this helps to clarify things for you.

**Comment by alexdowad** [comment]
> but they are different.

To make things (perhaps) a bit clearer for @DMA57361: you have `(-1).week` in one case, and `-(1.week)` in the other. This isn't just RuboCop, this is how Ruby's parser works.

**Comment by alexdowad** [comment]
This issue can be closed.
